### PR TITLE
Add io.github.tareksander.StoryReader

### PR DESCRIPTION
**Pull request by tareksander**
### Please confirm your submission meets all the criteria

- [x] Please describe the application briefly.

StoryReader is a reader and offline library for web novels, currently supporting the ScribbleHub and RoyalRoad sites.

- [x] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [x] I have read the and followed all the [Submission and licence requirements][reqs].
- [x] I have [built][build] and tested the submission locally.
- [x] I am an author/developer/upstream contributor of the project.

This release is technically not a "stable release", as I haven't hit 1.0 yet, but I don't expect any breaking changes or impossible database migrations to happen, so it should be forwards compatible with later versions.

This is a Flutter app, so I resorted to simply extracting a build from a GitHub release, as it seems to be the standard way of uploading Flutter apps on FlatHub.

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: io.github.tareksander.story_reader.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  "--share=network",
+  "--socket=pulseaudio",
+  "--socket=fallback-x11",
+  "--socket=wayland",
+  "--share=ipc",
+  "--device=dri"
+]
+cleanup: [
+  "/lib"
+]
```
Comment (hfiguiere) [inline io.github.tareksander.story_reader.yaml:16]: please use the proper yaml syntax.

File: io.github.tareksander.story_reader.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+modules:
+  - name: app
```
Comment (hfiguiere) [inline io.github.tareksander.story_reader.yaml:18]: please use a descriptive name here.

File: io.github.tareksander.story_reader.yaml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    buildsystem: simple
+    build-commands:
+      - install -Dm755 story_reader /app/bin/story_reader
+      - cp -r data /app/bin
+      - cp -r lib /app/bin
+      - mkdir -p /app/share/metainfo /app/share/applications /app/share/icons/hicolor/512x512/apps
+      - cp io.github.tareksander.story_reader.metainfo.xml /app/share/metainfo
+      - cp icon.png /app/share/icons/hicolor/512x512/apps/io.github.tareksander.story_reader.png
+      - cp io.github.tareksander.story_reader.desktop /app/share/applications
```
Comment (hfiguiere) [inline io.github.tareksander.story_reader.yaml:27]:
```suggestion
      - install -Dm644 io.github.tareksander.story_reader.metainfo.xml -t /app/share/metainfo
      - install -Dm644 icon.png /app/share/icons/hicolor/512x512/apps/io.github.tareksander.story_reader.png
      - install -Dm644 io.github.tareksander.story_reader.desktop -t /app/share/applications
```

File: io.github.tareksander.story_reader.desktop
Comment (hfiguiere) [inline io.github.tareksander.story_reader.desktop:1]: the release doesn't come with a desktop file or an icon? That's a bug.

If it does these then have no reason to be here.

File: io.github.tareksander.story_reader.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+finish-args: [
+  "--share=network",
+  "--socket=pulseaudio",
+  "--socket=fallback-x11",
+  "--socket=wayland",
+  "--share=ipc",
+  "--device=dri"
+]
+cleanup: [
+  "/lib"
```
Comment (hfiguiere) [inline io.github.tareksander.story_reader.yaml:15]: I don't see anything installed in `lib`.

File: flathub.json [hunk 5]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (hfiguiere) [inline flathub.json:2]: this remain disappointing.

File: io.github.tareksander.story_reader.desktop
Comment (tareksander) [reply io.github.tareksander.story_reader.desktop:1]: The release is just the Flutter Linux release bundle. But you're right, I could include the desktop file and icon there.

File: flathub.json (hunk 5 above)
Comment (tareksander) [reply flathub.json:2]: Flutter doesn't support cross-compiling and I haven't set up a GitHub Action for an aarch64 Linux build yet, so for now it's x86_64-only.

File: io.github.tareksander.story_reader.yaml (hunk 4 above)
Comment (tareksander) [reply io.github.tareksander.story_reader.yaml:15]: I saw something in `/lib` after the flatpak builder finished, I think it extracted debug symbols from the binaries or something? God if it's not needed.

File: io.github.tareksander.story_reader.desktop
Comment (hfiguiere) [reply io.github.tareksander.story_reader.desktop:1]: and the appstream file.

File: io.github.tareksander.story_reader.yaml (hunk 4 above)
Comment (hfiguiere) [reply io.github.tareksander.story_reader.yaml:15]: one doesn't delete the debug symbols.

File: io.github.tareksander.story_reader.desktop
Comment (tareksander) [reply io.github.tareksander.story_reader.desktop:1]: I thought that has to be in the repo? Is it extracted from the finished flatpak to show in the stores?

Comment (hfiguiere) [reply io.github.tareksander.story_reader.desktop:1]: no. a perfect submission only has a manifest.

File: io.github.tareksander.story_reader.yaml (hunk 2 above)
Comment (tareksander) [reply io.github.tareksander.story_reader.yaml]: Better now? As you may have noticed with the app name, I'm quite bad at naming things.

**Comment by tareksander** [comment]
I think I fixed all important things now.

File: io.github.tareksander.story_reader.yaml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+finish-args:
+- "--share=network"
+- "--socket=pulseaudio"
+- "--socket=fallback-x11"
+- "--socket=wayland"
+- "--share=ipc"
+- "--device=dri"
+
+modules:
+  - name: "x86_64_binary_bundle"
```
Comment (hfiguiere) [inline io.github.tareksander.story_reader.yaml:15]:
```suggestion
  - name: "story_reader"
```

Really naming things isn't hard.

**Comment by hfiguiere** [comment]
bot, build io.github.tareksander.story_reader

File: flathub.json (hunk 5 above)
Comment (tareksander) [reply flathub.json:2]: I checked just now, there actually aren't aarch64 Linux runners for GitHub Actions yet, but are planned for later this year. So until then there's nothing I can do if the FlatHub build process doesn't allow building from source with network access.

**Comment by tareksander** [comment]
I forgot the appid has to be the exact repo name, and for some reason the linter didn't complain for me. Fixed that now.

**Comment by bbhtt** [comment]
bot, build io.github.tareksander.story_reader

**Comment by bbhtt** [comment]
bot, build io.github.tareksander.StoryReader

**Comment by tareksander** [comment]
I missed the icon in the desktop file. Fixed that, too and ran the validation command that failed.

**Comment by bbhtt** [comment]
bot, build io.github.tareksander.StoryReader

**Comment by tareksander** [comment]
Works

**Pull request closed without merging**